### PR TITLE
chore(sdk): website sdk tests

### PR DESCRIPTION
**Pull request by tsuf239**
* testing website's initiation, `addJson`, and `url`
* add external `fetch` and `readFile` methods (js)

## Checklist

- [x] Title matches [Winglang's style guide](https://docs.winglang.io/contributors/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: examples/tests/sdk_tests/website/website.w [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+let indexFile = Util.readFile("./website/website/index.html");
+let otherFile = Util.readFile("./website/website/inner-folder/other.html");
+
+w.addJson("config.json", config);
+
+assert(w.path.endsWith("sdk_tests/website/website"));
+
+
+test "test" {
+    // doesn't work on tf-aws, we need to create an inflight method for retrieveing the url (there are no inflight properties)
```
Comment (Chriscbr) [inline examples/tests/sdk_tests/website/website.w:20]:
```suggestion
    // TODO doesn't work on tf-aws, we need to create an inflight method for retrieveing the url (there are no inflight properties)
```

Comment (Chriscbr) [inline examples/tests/sdk_tests/website/website.w:20]: I'm not sure I understand the comment - the `.url` field is defined in preflight but it should still be accessible in inflight, so we shouldn't need to add any new inflight methods or properties. Is the issue that the token is not getting resolved (similar to https://github.com/winglang/wing/issues/2232)?

File: examples/tests/sdk_tests/website/website.w [hunk 2]
```diff
@@ -0,0 +1,24 @@
+bring cloud;
+
+let w = new cloud.Website(path: "./website");
+let config = { json: 1 };
+
+class Util {
+    extern "../external/http.js" static inflight http(url: str): Json;
+    extern "../external/fs.js" static readFile(path: str): str;    
+  }
```
Comment (eladb) [inline examples/tests/sdk_tests/website/website.w:9]: Indentation

File: examples/tests/sdk_tests/website/website.w [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ 
+let indexFile = Util.readFile("./website/website/index.html");
+let otherFile = Util.readFile("./website/website/inner-folder/other.html");
+
+w.addJson("config.json", config);
+
+assert(w.path.endsWith("sdk_tests/website/website"));
```
Comment (eladb) [inline examples/tests/sdk_tests/website/website.w:16]: can you add a comment explaining what's being tested here (it seems like we are verifying that `path` is resolved to an absolute path, correct?)

File: examples/tests/sdk_tests/website/website.w (hunk 1 above)
Comment (tsuf239) [reply examples/tests/sdk_tests/website/website.w]: it's resolves well in sim and as undefined in aws... opened an issue #2829

File: examples/tests/sdk_tests/website/website/index.html
Comment (Chriscbr) [inline examples/tests/sdk_tests/website/website/index.html:1]: optional: If these files are empty, maybe they don't need to be saved in `git` (we can add their paths to .gitignore, and change the test so it creates the file during preflight)

**Review by Chriscbr — APPROVED** [review]

File: examples/tests/sdk_tests/website/website/index.html
Comment (tsuf239) [reply examples/tests/sdk_tests/website/website/index.html:1]: cool, I'll remember that for the next time :) 
not empty actually, just moved

**Pull request merged**